test/qos2.js

```diff
@@ -129,9 +129,10 @@ test('call published method with client with QoS 2', function (t) {
   }
 
   broker.published = function (packet, client, cb) {
-    t.ok(client, 'client must be passed to published method')
-
-    cb()
+    if (packet.topic.split('/').pop() !== 'clientDisconnect') {
```

Comment:
Yes. but we should put a note here, and maybe in the docs too. For server publishes the client is always null.